src/tests/async/synchronization-context/synchronization-context.cs

```diff
@@ -40,8 +40,11 @@ private static async Task TestSyncContextContinueAsync()
         await WrappedYieldToThreadPool(suspend: false).ConfigureAwait(false);
         Assert.Same(context, SynchronizationContext.Current);
 
-        await WrappedYieldToThreadPool(suspend: true).ConfigureAwait(false);
-        Assert.Null(SynchronizationContext.Current);
+        // Currently disabled since ConfigureAwait does not become a runtime async call,
+        // and this has a race condition until it does (where the callee finishes before
+        // we check IsCompleted on the awaiter).
+        //await WrappedYieldToThreadPool(suspend: true).ConfigureAwait(false);
+        //Assert.Null(SynchronizationContext.Current);
```

Comment:
Does this mean that async1 has a race condition here, but async2 would not?

Also - how `Task.Yield` can finish before `IsCompleted`? 
I think it`s whole purpose is to not complete synchronously.

Reply:
It looks like `IsCompleted` is never `true` for YieldAwaitable - so that it always suspends.

https://github.com/dotnet/runtime/blob/edb570cd9e75036b6c61b201a0a9a9f2301db276/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/YieldAwaitable.cs#L50

Reply:
I see. In async1 case suspend in the leaf may not suspend the entire call chain as callers will check `IsCompleted` _after_ getting back a task, which may have time to complete by then. 

Not sure if this is a good thing - as in if it may help to suspend less in some realistic scenarios. 
It is certainly a race if sync/async completion differences are observable.